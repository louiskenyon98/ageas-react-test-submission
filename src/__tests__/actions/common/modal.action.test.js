import React from 'react';
import {
    showSuccessModal,
    showFailModal,
    showDeleteNoteConfirmationModal
} from '../../../app/actions/common/modal.action';
import {
    SHOW_CONFIRMATION_MODAL,
    SHOW_STATUS_MODAL
} from '../../../app/actions/types';

let dispatch = jest.fn((objOrFunc) => {
    return typeof objOrFunc === 'function' ? objOrFunc(dispatch) : objOrFunc
});

const details = {
    title: 'Details Title',
    body: 'Details Body'
};

beforeEach(() => {
    jest.clearAllMocks();
});

describe('modal actions', () => {
    describe('showSuccessModal', () => {
        it('should dispatch SHOW_STATUS_MODAL with status of success', () => {
            const expectedAction = {
                type: SHOW_STATUS_MODAL,
                payload: {
                    status: 'success',
                    ...details
                }
            };
            dispatch(showSuccessModal(details));
            expect(dispatch).toHaveBeenCalledWith(expectedAction);
        });
    });
    describe('showFailModal', () => {
        it('should dispatch SHOW_STATUS MODAL with status of fail', () => {
            const expectedAction = {
                type: SHOW_STATUS_MODAL,
                payload: {
                    status: 'fail',
                    ...details
                }
            };
            dispatch(showFailModal(details));
            expect(dispatch).toHaveBeenCalledWith(expectedAction);
        });
    });
    describe('showDeleteNoteConfirmationModal', () => {
        it('should show confirmation modal', () => {
            const expected = {
                type: SHOW_CONFIRMATION_MODAL,
                payload: {
                    body: 'Do you want to delete this note',
                    accept: {
                        label: 'YES',
                        callback: expect.anything()
                    },
                    decline: {
                        label: 'NO',
                        callback: expect.anything()
                    }
                }
            };
            dispatch(showDeleteNoteConfirmationModal(91));
            expect(dispatch).toHaveBeenNthCalledWith(2, expected);
            expect(dispatch).toMatchSnapshot();
        });
    });
});
