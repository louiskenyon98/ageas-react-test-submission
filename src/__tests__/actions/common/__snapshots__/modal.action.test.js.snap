// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal actions showDeleteNoteConfirmationModal should show confirmation modal 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
    ],
    Array [
      Object {
        "payload": Object {
          "accept": Object {
            "callback": [Function],
            "label": "YES",
          },
          "body": "Do you want to delete this note",
          "decline": Object {
            "callback": [Function],
            "label": "NO",
          },
        },
        "type": "SHOW_CONFIRMATION_MODAL",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Object {
        "payload": Object {
          "accept": Object {
            "callback": [Function],
            "label": "YES",
          },
          "body": "Do you want to delete this note",
          "decline": Object {
            "callback": [Function],
            "label": "NO",
          },
        },
        "type": "SHOW_CONFIRMATION_MODAL",
      },
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
